### ISSUE TITLE
Cannot read parquet file from remote hdfs

### ISSUE THREAD

**Issue by lokinell** [dask/dask#2170]
My environment is below:
Python 3.5.3
Dask 0.14.1

My code like below:
```
from dask import dataframe as dd
path = "hdfs://192.168.3.244:8020/data_set/sample2/test.parquet"
df = dd.read_parquet(path)
```

When I try to run the code above, I get below exception:
```
Traceback (most recent call last):
  File "/Users/admin/anaconda3/lib/python3.5/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Users/admin/anaconda3/lib/python3.5/unittest/case.py", line 601, in run
    testMethod()
  File "/Users/admin/PycharmProjects/pdas-machine-learn/tests/test_api.py", line 36, in test_read_parquet
    df = dd.read_parquet(path)
  File "/Users/admin/anaconda3/lib/python3.5/site-packages/dask/dataframe/io/parquet.py", line 77, in read_parquet
    pf = fastparquet.ParquetFile(path, open_with=myopen, sep=myopen.fs.sep)
  File "/Users/admin/anaconda3/lib/python3.5/site-packages/fastparquet/api.py", line 52, in __init__
    with open_with(fn, 'rb') as f:
  File "/Users/admin/anaconda3/lib/python3.5/site-packages/dask/bytes/core.py", line 314, in __enter__
    f = f2 = self.myopen(self.path, mode=mode)
  File "/Users/admin/anaconda3/lib/python3.5/site-packages/distributed/hdfs.py", line 29, in open
    return HDFileSystem.open(self, path, mode, **kwargs)
  File "/Users/admin/anaconda3/lib/python3.5/site-packages/hdfs3/core.py", line 248, in open
    block_size=block_size)
  File "/Users/admin/anaconda3/lib/python3.5/site-packages/hdfs3/core.py", line 597, in __init__
    self._set_handle()
  File "/Users/admin/anaconda3/lib/python3.5/site-packages/hdfs3/core.py", line 607, in _set_handle
    (self.path, self.mode, msg))
OSError: Could not open file: 192.168.3.244:8020/data_set/sample2/test.parquet, mode: rb File does not exist: /user/admin/192.168.3.244:8020/data_set/sample2/test.parquet
	at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:71)
	at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:61)
	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocationsInt(FSNamesystem.java:1828)
	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:1799)
	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:1712)
	at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getBlockLocations(NameNodeRpcServer.java:672)
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getBlockLocations(ClientNamenodeProtocolServerSideTranslatorPB.java:373)
	at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:616)
	at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:969)
	at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2206)
	at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2202)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1709)
	at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2200)
```

I think it is not set right open for fastparquet, because I can use fastparquet directly with HDFileSystem open to read the same parquet file successfully.

**Comment by lokinell** [comment]
I think parquet.py should change like below:

```
   path = myopen.storage_options['path']
    try:
        pf = fastparquet.ParquetFile(path + myopen.fs.sep + '_metadata',
                                     open_with=myopen,
                                     sep=myopen.fs.sep)
```

**Comment by martindurant** [comment]
Fixed by PR #2171

**Comment by frol** [comment]
Fixed in #2207 and https://github.com/dask/distributed/pull/1014.
